### PR TITLE
DL-3199 Back Link Tests

### PR DESCRIPTION
**Pull request by Sir-Iyke**
DL-3199 Back Link Tests

Added back link tests so that they can be removed from the acceptance tests

## Checklist

*Reviewee* (Iyke)
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions
 - [x]  I've added my code using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

*Reviewer* (Replace with your name)
 - [ ]  I've confirmed that every effort has been made to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've confirmed appropriate tests has been included with any code added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions
 - [ ]  I've confirmed code was added using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: test/controllers/CorrespondenceAddressControllerSpec.scala [hunk 1]
```diff
@@ -78,6 +78,8 @@ class CorrespondenceAddressControllerSpec extends PlaySpec with GuiceOneServerPe
             document.getElementById("postcode").text() must be("")
             document.getElementById("country_field").text() must include("Country")
             document.getElementById("submit").text() must be("Continue")
+            document.getElementById("backLinkHref").text() must be("Back")
+            document.getElementById("backLinkHref").attr("href") must be("/ated-subscription/registered-business-address")
```
Comment (jeremystone) [inline test/controllers/CorrespondenceAddressControllerSpec.scala:82]: Also need to add test for other scenario (when editing existing data) e.g. by adding 
```
            document.getElementById("backLinkHref").text() must be("Back")
            document.getElementById("backLinkHref").attr("href") must be("/ated-subscription/review-business-details")
```
to the "if data exists in keystore, fill the form with that data in the page" test

File: test/controllers/RegisteredBusinessControllerSpec.scala [hunk 2]
```diff
@@ -81,6 +81,7 @@ class RegisteredBusinessControllerSpec extends PlaySpec with GuiceOneServerPerSu
             document.title() must be("Is this where you want us to send any letters about ATED? - GOV.UK")
             document.getElementById("business-registered-text").text() must be("This section is: ATED registration")
             document.getElementById("registered-business-address-header").text() must be("Is this where you want us to send any letters about ATED?")
+            document.getElementById("backLinkHref").text() must be("Back")
```
Comment (jeremystone) [inline test/controllers/RegisteredBusinessControllerSpec.scala:84]: Also check the back link here is what is in appconfig:
```suggestion
            document.getElementById("backLinkHref").text() must be("Back")
            document.getElementById("backLinkHref").attr("href") must be(backToBusinessCustomerUrl)
```
Requires mocking at the top of the spec...
```
  val backToBusinessCustomerUrl = "someBackToBusinessCustomerUrl"
  when(mockAppConfig.backToBusinessCustomerUrl).thenReturn(backToBusinessCustomerUrl)
```
Also might want to check the back link in the form validation scenarios.

File: test/controllers/CorrespondenceAddressControllerSpec.scala (hunk 1 above)
Comment (Sir-Iyke) [reply test/controllers/CorrespondenceAddressControllerSpec.scala:82]: I've added this

File: test/controllers/RegisteredBusinessControllerSpec.scala (hunk 2 above)
Comment (Sir-Iyke) [reply test/controllers/RegisteredBusinessControllerSpec.scala:86]: Which form validation scenarios where you referring to?

Comment (jeremystone) [reply test/controllers/RegisteredBusinessControllerSpec.scala:86]: Sorry - when it redisplays the page with validation errors. So scenarios: **return to this page if we have an error form validation** and maybe **return to this page if we have an error**

Comment (Sir-Iyke) [reply test/controllers/RegisteredBusinessControllerSpec.scala:86]: I have added these tests

**Review by jeremystone — APPROVED** [review]

**Pull request merged**